src/steadystateproblem.cpp

```diff
@@ -139,16 +140,16 @@ void SteadystateProblem::findSteadyStateByNewtonsMethod(Model *model,
     }
 }
 
-void SteadystateProblem::findSteadyStateBySimulation(const Solver *solver,
-                                                     Model *model, int it) {
+void SteadystateProblem::findSteadyStateBySimulation(const Solver &solver,
+                                                     Model &model, int it) {
     try {
         if (it < 0) {
             /* Preequilibration? -> Create a new solver instance for sim */
             bool integrateSensis = getSensitivityFlag(
                 model, solver, it, SteadyStateContext::solverCreation);
             auto newtonSimSolver = createSteadystateSimSolver(
                 solver, model, integrateSensis, false);
-            runSteadystateSimulation(newtonSimSolver.get(), model, false);
+            runSteadystateSimulation(*(newtonSimSolver.get()), model, false);
```

Comment:
```suggestion
            runSteadystateSimulation(*newtonSimSolver, model, false);
```

Can dereference a `std::unique_ptr` directly